### PR TITLE
windows: Adding osqueryi and osqueryd integration tests

### PR DESCRIPTION
**Pull request by muffins**
This ports the integration tests for osqueryi and osqueryd to Windows. Some of these tests still need a bit of work to get functioning the way we expect, however this should allow us to start building integration tests around the behavior of the daemon to address existing bugs.

### CODE REVIEW

**Comment by osqueryer** [comment]
:no_good: The commit cde987abf324d634a67a094f9a730b42049439c1 (Job results: [2026](https://jenkins.osquery.io/job/osqueryPullRequestAudit/2026/)) failed the code audit or documentation test.

**Comment by osqueryer** [comment]
:-1: The commit cde987abf324d634a67a094f9a730b42049439c1 (Job results: [1403](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1403/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit cde987abf324d634a67a094f9a730b42049439c1 (Job results: [5289](https://jenkins.osquery.io/job/osqueryPullRequestLinux/5289/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit cde987abf324d634a67a094f9a730b42049439c1 (Job results: [1405](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1405/)) failed one or more tests (**macOS/OS X**).

File: tools/tests/test_base.py [hunk 1]
```diff
@@ -288,21 +332,74 @@ def isDead(self, pid, timeout=5):
         return False
 
 
+'''Determine the last build osqueryi.exe'''
```
Comment (obelisk) [inline tools/tests/test_base.py]: nit, this is above the daemon check and uses a different comment style than the one below

File: tools/tests/test_osqueryi.py [hunk 2]
```diff
@@ -64,7 +67,12 @@ def test_config_dump(self):
             SHELL_TIMEOUT)
         content = ""
         with open(config, 'r') as fh: content = fh.read()
-        self.assertEqual(proc.stdout, "{\"%s\": %s}\n" % (config, content))
+        actual = proc.stdout
+
+        if os.name == "nt":
+            # Get rid of return carriages!
```
Comment (obelisk) [inline tools/tests/test_osqueryi.py]: nit carriage return

File: tools/tests/test_osqueryd.py [hunk 3]
```diff
@@ -84,8 +88,9 @@ def test_4_daemon_sighup(self):
         })
         self.assertTrue(daemon.isAlive())
 
-        # Send a SIGHUP
-        os.kill(daemon.proc.pid, signal.SIGHUP)
+        # Send a SIGHUP on posix, no SIGHUP on Windows.
```
Comment (obelisk) [inline tools/tests/test_osqueryd.py]: nit, SIGTERM on Windows

File: tools/tests/test_osqueryd.py [hunk 4]
```diff
@@ -136,7 +144,8 @@ def results_exists():
             self.assertTrue(os.path.exists(pth))
 
             # Only apply the mode checks to .log files.
-            if pth.find('.log') > 0:
+            # TODO: Add ACL checks for Windows logs
+            if pth.find('.log') > 0 and os.name != "nt":
```
Comment (obelisk) [inline tools/tests/test_osqueryd.py]: Is this globbing?

Comment (muffins) [inline tools/tests/test_osqueryd.py]: No it's just looking for the presence of `.log` in the given filename, I believe this is just a string search and not a regex based one.

File: tools/tests/test_base.py [hunk 5]
```diff
@@ -288,21 +332,72 @@ def isDead(self, pid, timeout=5):
         return False
 
 
+# Determine the last build osqueryd
+def getLatestOsqueryDaemon():
```
Comment (muffins) [inline tools/tests/test_base.py]: This and the `getLatestShell` functions can be merged into one.

File: tools/tests/winexpect.py [hunk 6]
```diff
@@ -0,0 +1,100 @@
...
+        self.child = proc
+        self.prompt = orig_prompt
+        self.prompt_change = prompt_change
+        self.continuation_prompt = continuation_prompt
+        self.timeout = timeout
+
+    # We currently only support 1 query at a time.
+    def run_command(self, command):
+        res = ''
+        if (command == ' '):
```
Comment (ryanheffernan) [inline tools/tests/winexpect.py]: I'd recommend something more like
command = command.strip()
if not command:

File: tools/tests/test_osqueryi.py [hunk 7]
```diff
@@ -64,7 +67,12 @@ def test_config_dump(self):
             SHELL_TIMEOUT)
         content = ""
         with open(config, 'r') as fh: content = fh.read()
-        self.assertEqual(proc.stdout, "{\"%s\": %s}\n" % (config, content))
+        actual = proc.stdout
+
+        if os.name == "nt":
+            # Get rid of carriage returns
+            actual = actual.replace("\r", "")
+        self.assertEqual(actual, "{\"%s\": %s}\n" % (config, content))
```
Comment (ryanheffernan) [inline tools/tests/test_osqueryi.py]: You can use single quotes to remove the need for escapes
'{"%S": %S}\n'

File: tools/tests/test_osqueryi.py [hunk 8]
```diff
@@ -64,7 +67,12 @@ def test_config_dump(self):
             SHELL_TIMEOUT)
         content = ""
         with open(config, 'r') as fh: content = fh.read()
-        self.assertEqual(proc.stdout, "{\"%s\": %s}\n" % (config, content))
+        actual = proc.stdout
+
+        if os.name == "nt":
+            # Get rid of carriage returns
+            actual = actual.replace("\r", "")
```
Comment (ryanheffernan) [inline tools/tests/test_osqueryi.py]: you can just do
```
actual = proc.stdout.strip()
```
and avoid the 'if'
```
>>> x
'adfadafda\r'
>>> x.strip()
'adfadafda'
```

**Comment by osqueryer** [comment]
:-1: The commit ac6eb85eac4f6d1799f5b7e4b20c2bccb77c6ec4 (Job results: [5303](https://jenkins.osquery.io/job/osqueryPullRequestLinux/5303/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit ac6eb85eac4f6d1799f5b7e4b20c2bccb77c6ec4 (Job results: [1418](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1418/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit ac6eb85eac4f6d1799f5b7e4b20c2bccb77c6ec4 (Job results: [2198](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2198/)) failed one or more tests (**Windows**).

**Comment by osqueryer** [comment]
:-1: The commit ac6eb85eac4f6d1799f5b7e4b20c2bccb77c6ec4 (Job results: [5304](https://jenkins.osquery.io/job/osqueryPullRequestLinux/5304/)) failed one or more tests (**Linux**).

**Comment by osqueryer** [comment]
:-1: The commit ac6eb85eac4f6d1799f5b7e4b20c2bccb77c6ec4 (Job results: [1419](https://jenkins.osquery.io/job/osqueryPullRequestOSX/1419/)) failed one or more tests (**macOS/OS X**).

**Comment by osqueryer** [comment]
:-1: The commit 4ad4adf49670196fecc0e8b2d729c371afd8ac4b (Job results: [2200](https://jenkins.osquery.io/job/osqueryPullRequestWindows/2200/)) failed one or more tests (**Windows**).

**Pull request merged**